examples/custom_rule.jl

```diff
@@ -1,5 +1,12 @@
 # # Enzyme custom rules tutorial
-
+#
+#
+# !!! note "More Examples"
+#     The tutorial below focuses on a simple setting to illustrate the basic concepts of writing custom rules.
+#     For more complex custom rules beyond the scope of this tutorial, you may take inspiration from the following in-the-wild examples:
+#     - [Enzyme internal rules](https://github.com/EnzymeAD/Enzyme.jl/blob/main/src/internal_rules.jl)
+#     - [LinearSolve.jl](https://github.com/SciML/LinearSolve.jl/blob/main/ext/LinearSolveEnzymeExt.jl)
```

Comment:
Also add https://github.com/FluxML/NNlib.jl/blob/master/ext/NNlibEnzymeCoreExt/NNlibEnzymeCoreExt.jl and KA https://github.com/JuliaGPU/KernelAbstractions.jl/blob/main/ext/EnzymeExt.jl ?

Reply:
Done